tmate-ssh-server.c

```diff
@@ -473,7 +495,28 @@ void tmate_ssh_server_main(struct tmate_session *session, const char *keys_dir,
 	if (!client->session)
 		tmate_fatal("Cannot initialize session");
 
+	fds[0].fd = ssh_bind_get_fd(daemon_bind);
+	fds[0].events = POLLIN;
+	fds[1].fd =  ssh_bind_get_fd(client_bind);
+	fds[1].events = POLLIN;
+
 	for (;;) {
+		ret = poll(fds, 2, 10 * 1000);
+
+		if (ret == 0 || ret < 0 && errno == EINTR)
+			continue;
+
```

Comment:
missing check if `ret < 0` and `errno != EINTR` (you can add parenthesis to the previous `if` for clarity)